### PR TITLE
Add config for the main gke jenkins jobs to e2e.sh

### PR DESCRIPTION
**Pull request by j3ffml**
Verified working. The gke-jenkins below are currently running from this script.

cc @ixdy, @quinton-hoole

### CODE REVIEW

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -93,6 +97,22 @@ GCE_DEFAULT_SKIP_TESTS=(
...
+# Tests which cannot be run on GKE, e.g. because they require
+# master ssh access.
+GKE_REQUIRED_SKIP_TESTS=(
+    "Nodes"
+    "Etcd\sFailure"
+    "MasterCerts"
+    "Shell"
+    )
+
+GKE_REBOOT_SKIP_TESTS=(
```
Comment (quinton-hoole) [inline hack/jenkins/e2e.sh]: It seems that these are almost all covered by GCE_DEFAULT_SKIP_TESTS which you already skip. Can you perhaps just delete GKE_REBOOT_SKIP_TESTS ?

Comment (quinton-hoole) [inline hack/jenkins/e2e.sh]: Oh, wait, I see that you don't currently skip GCE_DEFAULT_SKIP_TESTS in your reboot test.  Never mind.  You could potentially use --ginkgo-focus="Reboot" instead, but I don't think that's what you're trying to do here.  You're trying to run all the same tests as e2e-gce, and in addition "Reboot", right?  We should perhaps refactor all of this a bit to remove the duplication between GCE_DEFAULT_SKIP_TESTS and  GKE_REBOOT_SKIP_TESTS, as it seems that they always need to be identical with the exception of "Reboot"?

Comment (jlowdermilk) [inline hack/jenkins/e2e.sh]: Right, we want to run a small, unflaky set of tests along with reboot to exercise cluster functionality post-reboot. Refactored a bit to remove duplication.

File: hack/jenkins/e2e.sh [hunk 2]
```diff
@@ -496,6 +599,19 @@ ARTIFACTS=${WORKSPACE}/_artifacts
 mkdir -p ${ARTIFACTS}
 export E2E_REPORT_DIR=${ARTIFACTS}
 
+### Pre Set Up ###
```
Comment (quinton-hoole) [inline hack/jenkins/e2e.sh]: What's all this stuff for?  Please add a comment explaining why this is required.

**Comment by quinton-hoole** [comment]
Thanks @jlowdermilk .  One minor nit about a missing comment, then LGTM.

File: hack/jenkins/e2e.sh (hunk 2 above)
Comment (jlowdermilk) [inline hack/jenkins/e2e.sh]: Comment added.

**Comment by quinton-hoole** [comment]
LGTM.  Thanks @jlowdermilk

**Pull request merged**